File: code-javascript/net/socket_server_v1.js
const net = require("net");

const server = net.createServer((socket) => {
    console.log("Client connected");
    socket.on("data", (data) => {
        console.log("Received:", data.toString());
    });
    socket.write("Hello from Node.js Server");
    socket.end();
});

server.listen(8080, () => {
    console.log("Server listening on port 8080...");
});

File: code-python/net/socket_server_v1.py
import socket

server = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
server.bind(("0.0.0.0", 8080))
server.listen(1)
print("Server listening on port 8080...")

while True:
    conn, addr = server.accept()
    print("Client connected:", addr)
    data = conn.recv(1024)
    print("Received:", data.decode())
    conn.sendall(b"Hello from Python Server")
    conn.close()
